lib/cc/services/github_pull_requests.rb

```diff
@@ -73,4 +89,48 @@ def response_includes_repo_scope?(response)
   def test_status_code
     422
   end
+
+  def should_post_welcome_comment?
+    config.welcome_comment_enabled && @payload.fetch("first_contribution", false)
+  end
+
+  def welcome_comment_markdown_header
+    <<-HEADER
+Hey, @#{@payload.fetch("author_username")}-- Since this is the first PR we've seen from you,
+here's some things you should know about contributing to #{github_slug}:
+
+    HEADER
+  end
+
+  def admin_only_footer
+    <<-FOOTER
+
+* * *
+
+Quick note: By default, Code Climate will post the above comment on the *first* PR it sees from each contributor.
+If you'd like to customize this message or disable this, go [here](#{@payload.fetch("pull_request_integration_edit_url")}).
+    FOOTER
+  end
+
+  def welcome_comment_markdown
+    header = welcome_comment_markdown_header
+    body = config.welcome_comment_markdown
+    if @payload.fetch("author_is_site_admin")
+      header + body + admin_only_footer
+    else
+      header + body
+    end
+  end
+
+  def post_welcome_comment
+    pull_request_number = @payload.fetch("number")
+    comments_url = "#{config.base_url}/repos/#{github_slug}/issues/#{pull_request_number}/comments"
+
+    params = { body: welcome_comment_markdown }
+
+    # FIXME: is this appropriate? when we receive the pending event, we used to just update the status for the commit,
```

Comment:
That sounds reasonable to me.

In the event of one failure/one pass will the merged metadata be sufficient to know which is which?

> On Aug 24, 2016, at 16:44, Max Jacobson <notifications@github.com> wrote:
> 
> In lib/cc/services/github_pull_requests.rb:
> 
> > +    header = welcome_comment_markdown_header
> > +    body = config.welcome_comment_markdown
> > +    if @payload.fetch("author_is_site_admin")
> > +      header + body + admin_only_footer
> > +    else
> > +      header + body
> > +    end
> > +  end
> > +
> > +  def post_welcome_comment
> > +    pull_request_number = @payload.fetch("number")
> > +    comments_url = "#{config.base_url}/repos/#{github_slug}/issues/#{pull_request_number}/comments"
> > +
> > +    params = { body: welcome_comment_markdown }
> > +
> > +    # FIXME: is this appropriate? when we receive the pending event, we used to just update the status for the commit,
> OK my new plan is to merge the two network responses into one service event that reflects the success of both. It will be ok: true when both requests succeeded, and it will have the metadata from both responses for debugging purposes
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>